test/e2e/storage/regional_pd.go

```diff
@@ -0,0 +1,441 @@
+			Selector:  labels,
+		},
+	}
+
+	sts = &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "repd-sts",
+			Namespace: ns,
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			ServiceName:          svc.Name,
+			Replicas:             &replicas,
+			Template:             *newPodTemplate(labels),
+			VolumeClaimTemplates: []v1.PersistentVolumeClaim{*claimTemplate},
+		},
+	}
+
+	return
+}
+
+func newPodTemplate(labels map[string]string) *v1.PodTemplateSpec {
+	return &v1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: labels,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				// This container writes its pod name to a file in the Regional PD
+				// and prints the entire file to stdout.
+				{
+					Name:    "busybox",
+					Image:   "gcr.io/google_containers/busybox",
+					Command: []string{"sh", "-c"},
+					Args: []string{
+						"echo ${POD_NAME} >> /mnt/data/repd/pods.txt;" +
+							"cat /mnt/data/repd/pods.txt;" +
+							"sleep 3600;",
+					},
+					Env: []v1.EnvVar{{
+						Name: "POD_NAME",
+						ValueFrom: &v1.EnvVarSource{
+							FieldRef: &v1.ObjectFieldSelector{
+								FieldPath: "metadata.name",
+							},
+						},
+					}},
+					Ports: []v1.ContainerPort{{
+						ContainerPort: 80,
+						Name:          "web",
+					}},
+					VolumeMounts: []v1.VolumeMount{{
+						Name:      "repd-vol",
+						MountPath: "/mnt/data/repd",
+					}},
+				},
+			},
+		},
+	}
+}
+
+func newClaimTemplate(ns string) *v1.PersistentVolumeClaim {
+	return &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "repd-vol",
+			Namespace: ns,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				v1.ReadWriteOnce,
+			},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceName(v1.ResourceStorage): resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+}
+
+func newRegionalStorageClass(namespace string, zones []string) *storage.StorageClass {
+	return &storage.StorageClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "StorageClass",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace + "-sc",
+		},
+		Provisioner: "kubernetes.io/gce-pd",
+		Parameters: map[string]string{
+			"type":             "pd-standard",
+			"zones":            strings.Join(zones, ","),
+			"replication-type": "repd",
+		},
+	}
+}
+
+func getRandomCloudZones(c clientset.Interface) []string {
+	zones := framework.GetClusterZones(c)
+
+	Expect(zones.Len()).To(BeNumerically(">=", 2),
+		"The test should only be run in multizone clusters.")
+
+	zone1, _ := zones.PopAny()
+	zone2, _ := zones.PopAny()
+	return []string{zone1, zone2}
+}
+
+// Waits for at least 1 replica of a StatefulSet to become not ready or until timeout occurs, whichever comes first.
+func waitForStatefulSetReplicasNotReady(statefulSetName, ns string, c clientset.Interface) error {
```

Comment:
Have you checked if there are already existing statefulset utility functions?  I leveraged a few for a local PV test.